src/planning/include/planning/DijkstraPlanner.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PLANNING__DIJKSTRAPLANNER_HPP_
+#define PLANNING__DIJKSTRAPLANNER_HPP_
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <chrono>
+#include "planning/ComputePathToPoseTaskServer.hpp"
+#include "planning/Navfn.hpp"
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/point.hpp"
+#include "nav2_msgs/msg/costmap.hpp"
+#include "nav2_msgs/srv/get_costmap.hpp"
+
+class DijkstraPlanner : public ComputePathToPoseTaskServer
+{
+public:
+  explicit DijkstraPlanner(const std::string & name);
+  DijkstraPlanner() = delete;
+  ~DijkstraPlanner();
+
+  TaskStatus executeAsync(const ComputePathToPoseCommand::SharedPtr command) override;
+
+private:
+  // Compute a plan given start and goal poses, provided in global world frame.
+  bool makePlan(const geometry_msgs::msg::PoseStamped& start, const geometry_msgs::msg::PoseStamped& goal,
+    double tolerance, std::vector<geometry_msgs::msg::PoseStamped>& plan);
+
+  // Compute the navigation function given a seed point in the world to start from
+  bool computePotential(const geometry_msgs::msg::Point& world_point);
+
+  // Compute a plan to a goal from a potential - must call computePotential first
+  bool getPlanFromPotential(const geometry_msgs::msg::PoseStamped& goal, std::vector<geometry_msgs::msg::PoseStamped>& plan);
```

Comment:
Please run (all files) through ament_cpplint and ament_uncrustify. Several lines are too long.